### PR TITLE
Refactor: improvements to PanelQueryRunner

### PR DESCRIPTION
**Pull request by ryantxu**
Small steps towards supporting streaming:

1. merged `DataQueryOptions` + `DataRequestInfo` info: `DataQueryRequest`
2. add a `requestId` to every request
3. adds a `destroy()` function to `PanelQueryRunner` that gets called from PanelModel.destroy().  This now tries to cancel any open http requets with getBackendSrv()
4. Removes `rangeRaw` from the DataQueryRequest type.  The property is still set on the object via:
   ```
   (request as any).rangeRaw = timeRange.raw;
   ```
5. Pass `string|DataSourceApi` as datasource parameter to panelQueryRunner (previously two options)

### CODE REVIEW

**Comment by ryantxu** [comment]
@torkelo -- here are some minor changes that will make other issues easier to look at

File: public/app/features/dashboard/state/PanelQueryRunner.ts [hunk 1]
```diff
@@ -121,6 +126,8 @@ export class PanelQueryRunner {
       cacheTimeout,
       startTime: Date.now(),
     };
+    // Deprecated
+    (request as any).rangeRaw = timeRange.raw;
```
Comment (torkelo) [inline public/app/features/dashboard/state/PanelQueryRunner.ts]: smart way to keep old deprecated stuff away from public interfaces 👍

File: public/app/features/dashboard/state/PanelQueryRunner.ts [hunk 2]
```diff
@@ -134,7 +141,10 @@ export class PanelQueryRunner {
     let loadingStateTimeoutId = 0;
 
     try {
-      const ds = options.ds ? options.ds : await getDatasourceSrv().get(datasource, request.scopedVars);
+      const ds =
+        datasource && (datasource as any).query
```
Comment (torkelo) [inline public/app/features/dashboard/state/PanelQueryRunner.ts]: is ds not types already?

**Review by torkelo — APPROVED** [review]
Looks good

**Pull request merged**